src/Catalyst.Core.Lib.Tests/IntegrationTests/FileSystem/FileStoreTest.cs

```diff
@@ -0,0 +1,196 @@
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Core.Lib.FileSystem;
+using Catalyst.TestUtils;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.Core.Lib.Tests.IntegrationTests.FileSystem
+{
+    public sealed class FileStoreTest : FileSystemBasedTest
+    {
+        public FileStoreTest(ITestOutputHelper output) : base(output) { }
+
+        private sealed class Entity
+        {
+            public int Number;
+            public string Value;
+        }
+
+        private Entity a = new Entity {Number = 1, Value = "a"};
+        private Entity b = new Entity {Number = 2, Value = "b"};
+
+        private FileStore<int, Entity> Store
+        {
+            get
+            {
+                var folder = Path.Combine(FileSystem.GetCatalystDataDir().FullName, "test-filestore");
+                if (!Directory.Exists(folder))
+                {
+                    Directory.CreateDirectory(folder);
+                }
+
+                return new FileStore<int, Entity>
+                {
+                    Folder = folder,
+                    NameToKey = name => name.ToString(),
+                    KeyToName = int.Parse
+                };
+            }
+        }
+
+        [Fact]
+        public async Task PutAndGet()
+        {
+            var store = Store;
+
+            await store.PutAsync(a.Number, a);
+            await store.PutAsync(b.Number, b);
+
+            var a1 = await store.GetAsync(a.Number);
+            Assert.Equal(a.Number, a1.Number);
+            Assert.Equal(a.Value, a1.Value);
+
+            var b1 = await store.GetAsync(b.Number);
+            Assert.Equal(b.Number, b1.Number);
+            Assert.Equal(b.Value, b1.Value);
+        }
+
+        [Fact]
+        public async Task TryGet()
+        {
+            var store = Store;
+            await store.PutAsync(3, a);
+            var a1 = await store.GetAsync(3);
+            Assert.Equal(a.Number, a1.Number);
+            Assert.Equal(a.Value, a1.Value);
+
+            var a3 = await store.TryGetAsync(42);
+            Assert.Null(a3);
+        }
+
+        [Fact]
+        public void Get_Unknown()
+        {
+            ExceptionAssert.Throws<KeyNotFoundException>(() =>
+            {
+                var _ = Store.GetAsync(42).Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)